src/bun.js/webcore/blob.zig

```diff
@@ -3794,6 +3794,10 @@ pub const Blob = struct {
                         } else {
                             return build.blob.dupe();
                         }
+                    } else if (current.toSliceOrNull(global)) |sliced| {
+                        if (sliced.allocator.get()) |allocator| {
+                            return Blob.initWithAllASCII(bun.constStrToU8(sliced.slice()), allocator, global, false);
```

Comment:
This should be cloned. There are too many cases where we assume Blob is always owned by mimalloc, but in this case it might not be.

Reply:
also if you do .arrayBuffer() on a Response it will transfer the buffer which would mean that the string is now mutable which will break an unknown number of things.

Reply:
weird `toSliceOrNull` dont clone it? 
https://github.com/oven-sh/bun/blob/main/src/bun.js/bindings/bindings.zig#L4193-L4198

Reply:
toSlice does not if it's an ASCII latin1 string. toOwnedSlice clones it.

Reply:
fixed on https://github.com/oven-sh/bun/pull/4127/commits/ef6989f4118ee842adf88f83c9d2b3ba64f04f0d